### PR TITLE
236493 add filter pipeline element

### PR DESCRIPTION
**Pull request by JBettonvil**
Adds the new Filter Pipeline Element into the submission pipeline, allowing filtering of incoming submissions according to user criteria.

### CODE REVIEW

**Review by ahofman — CHANGES_REQUESTED** [review]
Can you update the quickstart instructions in the README.md to include the required changes to the manifest?

File: Source/ReferenceConnectorService/ReferenceConnectorService.csproj [hunk 1]
```diff
@@ -15,7 +15,7 @@
     <PackageReference Include="Microsoft.ServiceFabric" Version="6.2.301" />
     <PackageReference Include="Microsoft.ServiceFabric.AspNetCore.Kestrel" Version="3.1.301" />
     <PackageReference Include="Microsoft.ServiceFabric.Services.Remoting" Version="3.1.301" />
-    <PackageReference Include="RecordPoint.Connectors.SDK" Version="0.0.0-CI-20181011-002554" />
+    <PackageReference Include="RecordPoint.Connectors.SDK" Version="0.0.0-CI-20190103-002229" />
```
Comment (ahofman) [inline Source/ReferenceConnectorService/ReferenceConnectorService.csproj]: This version of the SDK isn't published on NuGet, so this won't compile outside of RecordPoint. I can publish that version if we need to

File: Source/ReferenceConnectorWorkerService/ConnectorTask.cs [hunk 2]
```diff
@@ -154,20 +154,24 @@ private async Task SubmitItem(CancellationToken cancellationToken)
...
                 // Dead-letter the item to a durable data store where it can be retried later. (e.g., a message broker).
             }
-            
-            // After submitting, check to see if Records365 vNext has a record of the parent item that was referenced
-            // in the submission above. If it doesn't, we need to submit it.
-            if (submitContext.AggregationFoundDuringItemSubmission.HasValue &&
-                !submitContext.AggregationFoundDuringItemSubmission.Value)
+
+            //If filtered out by the FilterPipelineElement, we do not want to submit
+            if (submitContext.SubmitResult.SubmitStatus != SubmitResult.Status.Skipped)
```
Comment (ahofman) [inline Source/ReferenceConnectorWorkerService/ConnectorTask.cs]: Maybe just check `if (submitContext.SubmitResult.SubmitStatus == SubmitResult.Status.OK)` here instead?

File: Source/ReferenceConnectorWorkerService/ConnectorTask.cs [hunk 3]
```diff
@@ -154,20 +154,25 @@ private async Task SubmitItem(CancellationToken cancellationToken)
                 // Something went wrong trying to submit the item. 
                 // Dead-letter the item to a durable data store where it can be retried later. (e.g., a message broker).
             }
-            
-            // After submitting, check to see if Records365 vNext has a record of the parent item that was referenced
-            // in the submission above. If it doesn't, we need to submit it.
-            if (submitContext.AggregationFoundDuringItemSubmission.HasValue &&
-                !submitContext.AggregationFoundDuringItemSubmission.Value)
+
```
Comment (ahofman) [inline Source/ReferenceConnectorWorkerService/ConnectorTask.cs:158]: I'm pretty sure we need to pull the `Filters` off the ConnectorConfig and put it on the `SubmitContext` as well - I think @0shi can confirm

File: Source/ReferenceConnectorWorkerService/ConnectorTask.cs (hunk 2 above)
Comment (JBettonvil) [reply Source/ReferenceConnectorWorkerService/ConnectorTask.cs]: Done.

File: Source/ReferenceConnectorService/ReferenceConnectorService.csproj (hunk 1 above)
Comment (JBettonvil) [reply Source/ReferenceConnectorService/ReferenceConnectorService.csproj]: As there's a new version likely to arrive shortly, I will update this once the new version is done and published.

**Review by ahofman — APPROVED** [review]

**Pull request merged**